src/site-apps.ts

```diff
@@ -83,32 +83,28 @@ export class SiteApps {
     }
 
     setupSiteApp(loader: SiteAppLoader) {
-        const app: SiteApp = {
-            id: loader.id,
-            loaded: false,
-            errored: false,
-        }
-        this.apps[loader.id] = app
-
-        const onLoaded = (success: boolean) => {
-            this.apps[loader.id].errored = !success
-            this.apps[loader.id].loaded = true
-
-            logger.info(`Site app with id ${loader.id} ${success ? 'loaded' : 'errored'}`)
-
-            if (success && this.bufferedInvocations.length) {
+        const app = this.apps[loader.id]
+        const processBufferedEvents = () => {
+            if (!app.errored && this.bufferedInvocations.length) {
                 logger.info(`Processing ${this.bufferedInvocations.length} events for site app with id ${loader.id}`)
                 this.bufferedInvocations.forEach((globals) => app.processEvent?.(globals))
+                app.processedBuffer = true
             }
 
-            for (const app of Object.values(this.apps)) {
-                if (!app.loaded) {
-                    // If any other apps are not loaded, we don't want to stop buffering
-                    return
-                }
+            if (Object.keys(this.apps).every((id) => this.apps[id].processedBuffer || this.apps[id].errored)) {
+                this.stopBuffering?.()
             }
```

Comment:
Why iterate over the keys and then get the value?
```suggestion
            if (Object.values(this.apps).every((app) => app.processedBuffer || app.errored)) {
                this.stopBuffering?.()
            }
```

Reply:
Good spot, this was leftover from a previous version